frontend/simple/js/state.js

```diff
@@ -69,25 +94,45 @@ const mutations = {
     index === -1 && state.pending.push(contractId)
   }
 }
-
 // https://vuex.vuejs.org/en/getters.html
 // https://vuex.vuejs.org/en/modules.html
 const getters = {
   currentGroup (state) {
     return state[state.currentGroupId]
+  },
+  mailbox (state) {
+    return getMailbox(state)
+  },
+  unread (state) {
+    let mailbox = getMailbox(state)
+    return mailbox && state[ mailbox ].messages.filter((msg) => !msg.read).length
   }
 }
 
 const actions = {
+  // Used to update contracts to the current state that the server is aware of
+  async synchronize ({dispatch}: {dispatch: Function}, contractId: string) {
```

Comment:
`synchronize`, again, is a very unclear name. Synchronize what? The entire app? The database? Try `syncContract`, or `syncContractWithServer` or `updateContractFromServer`.